include/swift/AST/TypeOrExtensionDecl.h

```diff
@@ -0,0 +1,59 @@
+//===- TypeOrExtensionDecl.h - Swift Language Declaration ASTs -*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the TypeOrExtensionDecl struct, separately to Decl.h so
+// that this can be included in files that Decl.h includes.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_TYPE_OR_EXTENSION_DECL_H
+#define SWIFT_TYPE_OR_EXTENSION_DECL_H
+
+#include "llvm/ADT/PointerUnion.h"
+
+namespace swift {
+class Decl;
+class NominalTypeDecl;
+class ExtensionDecl;
+class DeclContext;
+
+/// \brief Describes either a nominal type declaration or an extension
+/// declaration.
+struct TypeOrExtensionDecl {
+  // (The definitions are in Decl.cpp.)
+  llvm::PointerUnion<NominalTypeDecl *, ExtensionDecl *> Decl;
+
+  TypeOrExtensionDecl() = default;
+
+  TypeOrExtensionDecl(NominalTypeDecl *D);
+  TypeOrExtensionDecl(ExtensionDecl *D);
+
+  /// \brief Return the contained *Decl as the Decl superclass.
+  class Decl *getAsDecl() const;
```

Comment:
Stray `class` here.

Reply:
The field is called `Decl`: clang specifically said I should write `class Decl` to disambiguate.

Reply:
Ah, I guess I would have just renamed the field.